File: frontend/src/views/file/fileParseStatus.ts
export type FileParseStatusLike = { status?: string };

const statusText: Record<string, string> = {
  PENDING: '待处理',
  PARSING: '解析中',
  RUNNING: '解析中',
  PARSED: '解析成功',
  SUCCESS: '解析成功',
  FAILED: '解析失败',
  CANCELED: '已取消'
};

export function normalizeFileParseStatus(status?: string) {
  return (status || '').toUpperCase();
}

export function fileParseStatusText(status?: string) {
  const normalized = normalizeFileParseStatus(status);
  return statusText[normalized] || status || '未知';
}

export function hasActiveFileParse(records: FileParseStatusLike[]) {
  return records.some((record) => ['PENDING', 'PARSING', 'RUNNING'].includes(normalizeFileParseStatus(record.status)));
}

export function isRetryableFileParse(record: FileParseStatusLike) {
  return ['FAILED', 'CANCELED'].includes(normalizeFileParseStatus(record.status));
}

File: src/main/java/com/xd/smartworksite/file/domain/FileParseStatus.java
package com.xd.smartworksite.file.domain;

import java.util.Locale;
import java.util.Set;

public enum FileParseStatus {
    PENDING,
    PARSING,
    PARSED,
    RUNNING,
    SUCCESS,
    FAILED,
    CANCELED;

    private static final Set<String> ACTIVE = Set.of(PENDING.name(), PARSING.name(), RUNNING.name());
    private static final Set<String> PARSED_STATUSES = Set.of(PARSED.name(), SUCCESS.name());
    private static final Set<String> RETRYABLE = Set.of(FAILED.name(), CANCELED.name());

    public static boolean isActive(String status) {
        return ACTIVE.contains(normalize(status));
    }

    public static boolean isParsed(String status) {
        return PARSED_STATUSES.contains(normalize(status));
    }

    public static boolean isRetryable(String status) {
        return RETRYABLE.contains(normalize(status));
    }

    private static String normalize(String status) {
        return status == null ? "" : status.trim().toUpperCase(Locale.ROOT);
    }
}
